### PR TITLE
Fix `bigint` marshal, unmarshall

### PR DESCRIPTION
**Pull request by illia-li**
Fixed for `bigint` type:
1. `string` marshal and unmarshal as `nullable`. 
1.1. Before: marshals `""` - caused an error; unmarshals `nil data` into `"0"`.
1.2. Now: marshals and unmarshals `nil data` - `""`.
2. `custom string` marshals and unmarshals, before `not supported`, now `supported`.
3. `big.Int` marshals, before `not supported`, now `supported`.
4. Marshal data, which equal `math.MaxUint64`, into `uint64`, `uint`, return `error' before, now `no error`.
5. Unmarshall `data` with the length is different from `0` and `8`, did not return `error` before, now it returns `error'.

Close issues for `bigint`:
1) #246 
2) #243
3) #244
4) #250
5) #252

### CODE REVIEW

File: marshal_5_bigint_test.go [hunk 1]
```diff
@@ -7,178 +7,167 @@ import (
 	"github.com/gocql/gocql"
 	"github.com/gocql/gocql/internal/tests/serialization"
 	"github.com/gocql/gocql/internal/tests/serialization/mod"
+	bigint "github.com/gocql/gocql/marshal/bigint"
```
Comment (sylwiaszunejko) [inline marshal_5_bigint_test.go:10]: same as before

**Review by sylwiaszunejko — CHANGES_REQUESTED** [review]

File: marshal/bigint/marshal_utils.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
+package smallint
```
[the remark this answers is not in the document]
Comment (illia-li) [reply marshal/bigint/marshal_utils.go]: Fixed.

File: marshal/bigint/marshal.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
+package smallint
```
[the remark this answers is not in the document]
Comment (illia-li) [reply marshal/bigint/marshal.go]: Fixed.

File: marshal/bigint/unmarshal.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
+package smallint
```
[the remark this answers is not in the document]
Comment (illia-li) [reply marshal/bigint/unmarshal.go]: Fixed.

File: marshal/bigint/unmarshal_utils.go [hunk 5]
```diff
@@ -0,0 +1,469 @@
+package smallint
```
[the remark this answers is not in the document]
Comment (illia-li) [reply marshal/bigint/unmarshal_utils.go]: Fixed.

File: marshal_5_bigint_fails_test.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
+package gocql_test
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/gocql/gocql"
+	"github.com/gocql/gocql/internal/tests/serialization"
+	"github.com/gocql/gocql/internal/tests/serialization/mod"
+	bigint "github.com/gocql/gocql/marshal/bigint"
```
[the remark this answers is not in the document]
Comment (illia-li) [reply marshal_5_bigint_fails_test.go]: Fixed.

**Comment by illia-li** [comment]
Please take a look again

**Review by sylwiaszunejko — APPROVED** [review]
for me it LGTM, @dkropachev could you take a look?

File: marshal.go [hunk 7]
```diff
@@ -562,7 +584,7 @@ func marshalVarint(info TypeInfo, value interface{}) ([]byte, error) {
 			binary.BigEndian.PutUint64(retBytes, v)
 		}
 	default:
-		retBytes, err = marshalBigInt(info, value)
+		retBytes, err = marshalCounter(info, value)
```
[the remark this answers is not in the document]
Comment (illia-li) [reply marshal.go:587]: I didn't see these changes.
Renamed into `marshalBigIntOld`.

**Review by dkropachev — APPROVED** [review]

**Pull request merged**